geostore/check_stac_metadata/utils.py

```diff
@@ -137,6 +137,7 @@ def process_metadata(self) -> None:
                 hash_key=self.hash_key,
                 range_key=f"{ProcessingAssetType.METADATA.value}{DB_KEY_SEPARATOR}{index}",
                 url=metadata_file[PROCESSING_ASSET_URL_KEY],
+                filename=basename(str(metadata_file[PROCESSING_ASSET_URL_KEY])),
```

Comment:
You can avoid the `str()` conversion by pulling out the value and doing `assert isinstance(asset_url, str)`. Assertions are stripped out when optimising Python, so it shouldn't affect production speed.